switchboard/s3_util.py

```diff
@@ -0,0 +1,205 @@
+from d4s2_api.models import S3Delivery, EmailTemplate, S3User, S3UserTypes
+from d4s2_api.utils import ProcessedMessage, DeliveryMessage
+import boto3
+
+
+class S3DeliveryUtil(object):
+    def __init__(self, s3_delivery, user):
+        self.s3_delivery = s3_delivery
+        self.endpoint = s3_delivery.bucket.endpoint
+        self.source_bucket_name = s3_delivery.bucket.name
+        self.user = user
+        self._s3_agent = None
+        self._current_s3_user = None
+        self._destination_bucket_name = None
+
+    @property
```

Comment:
Can you clarify the reasons for using lazily-loaded properties here? It looks like everything is local (e.g. from the database) and known at time of `__init__`